### PR TITLE
[Network] `az network application-gateway `: Add preview command group 'client-cert' and 'ssl-profile'

### PR DESCRIPTION
**Pull request by msyyc**
**Description**

https://github.com/Azure/azure-cli/issues/16144
Application Gateway Mutual Authentication Support

**Testing Guide**

**History Notes**

[Network] `az network application-gateway  create`: Add parameter `--ssl-profile-id`
[Network] `az network application-gateway  client-cert`: Manage trusted client certificate of application gateway
[Network] `az network application-gateway  ssl-profile`:  Manage ssl profiles of application gateway

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/_actions.py [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 import argparse
 from knack.util import CLIError
-
+from ._validators import read_base_64_file
+from azure.cli.core.azclierror import UserFault
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_actions.py:11]: https://github.com/Azure/azure-cli/blob/5e3cd5cbd1fb4959e936bd832f44181584c33fe5/src/azure-cli-core/azure/cli/core/azclierror.py#L73

File: src/azure-cli/azure/cli/command_modules/network/_actions.py [hunk 2]
```diff
@@ -60,3 +61,70 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
...
+            kl = k.lower()
+            v = properties[k]
+            if kl == 'name':
+                d['name'] = v[0]
+            elif kl == 'min-protocol-version':
+                d['policy_type'] = v[0]
+            elif kl == 'min-protocol-version':
+                d['min_protocol_version'] = v[0]
+            elif kl == 'cipher-suites':
+                d['cipher-suites'] = _split(v[0])
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_actions.py:120]: cipher_suites ?

File: src/azure-cli/azure/cli/command_modules/network/_actions.py [hunk 3]
```diff
@@ -60,3 +61,70 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
...
+            elif kl == 'disabled-ssl-protocols':
+                d['disabled-ssl-protocols'] = _split(v[0])
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_actions.py:122]: 'disabled_ssl_protocols' ?

File: src/azure-cli/azure/cli/command_modules/network/_template_builder.py [hunk 4]
```diff
@@ -269,6 +274,49 @@ def _ag_subresource_id(_type, name):
     if firewall_policy and cmd.supported_api_version(min_api='2018-12-01'):
         ag_properties.update({'firewallPolicy': {'id': firewall_policy}})
 
+    # mutual authentication support
+    if cmd.supported_api_version(min_api='2020-06-01') and trusted_client_certificates:
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_template_builder.py:278]: Already define min_api in argument, so no need to add here.
```
 c.argument('ssl_profile', min_api='2020-06-01', nargs='+', action=SslProfilesCreate, is_preview=True)
```

File: src/azure-cli/azure/cli/command_modules/network/_template_builder.py [hunk 5]
```diff
@@ -269,6 +274,49 @@ def _ag_subresource_id(_type, name):
...
+                    "name": item['name'],
+                    "properties": {
+                        "data": item['data']
+                    }
+                }
+            )
+        ag_properties.update({"trustedClientCertificates": parameters})
+
+    # ssl profiles
+    if cmd.supported_api_version(min_api='2020-06-01') and ssl_profile:
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_template_builder.py:292]: Same concern.

**Comment by yonzhan** [comment]
Network

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**